App.js

```diff
@@ -6,109 +6,403 @@
+    }
+    else{
+      operacionAcontinuar = tipo_de_operacion
+      this.ResultadoOperacion(1)  
+    }
+  };
+
+  ResultadoOperacion=(CotinuarOperacion)=>{
+    let numero1 = parseFloat(PrimerNumero);
+    let numero2 = parseFloat(SegundoNumero);
+    let resultadoFinal = 0;
+    switch (operacionEnCurso) {
+      case '+':
+        resultadoFinal = numero1 + numero2
+        break;
+      case '-':
+        resultadoFinal = numero1 - numero2
+        break;
+      case '/':
+        resultadoFinal = numero1 / numero2
+        break;
+      case 'x':
+        resultadoFinal = numero1 * numero2
+        break;   
+    }
+    if(CotinuarOperacion){
+      SegundoNumero = '';
+      PrimerNumero = resultadoFinal.toString();
+      operacionEnCurso = operacionAcontinuar 
+      this.setState({
+        textoResultado: resultadoFinal,
+        textoOperacion: PrimerNumero + ' ' + operacionEnCurso + ' ',
+      })
+      
+    }
+    else{
+      PrimerNumero = '';
+      SegundoNumero = '';
+      operacionEnCurso='';
+      this.setState({
+        textoResultado: resultadoFinal,
+        textoOperacion : '',
+      })
+
+    }
+    
+  };
+
+  LimpiarPantalla(){
+    PrimerNumero = '';
+    SegundoNumero = '';
+    operacionEnCurso='';
+    this.setState({
+      textoResultado: '0',
+      textoOperacion : '',
+    })
+  }
+
+  BorrarUltimoCaracter(){
+    if(!PrimerNumero){
+
+    }
+    else if(PrimerNumero && !operacionEnCurso){
+      PrimerNumero = PrimerNumero.slice(0,-1);
+      this.setState({
+        textoOperacion : this.state.textoOperacion.slice(0,-1),
+      });
+    }
+    else if(operacionEnCurso && !SegundoNumero){
+      operacionEnCurso = '';
+      this.setState({
+        textoOperacion : PrimerNumero
+      });
+    }
+    else{
+      SegundoNumero = SegundoNumero.slice(0,-1);
+      this.setState({
+        textoOperacion : this.state.textoOperacion.slice(0,-1),
+      });
+
+    }
+  }
+
+  CambiarSigno(){
+    if(!PrimerNumero || (PrimerNumero && operacionEnCurso && !SegundoNumero)){
+    }
+    else{
+      let numeroSigno = 0;
+      if(PrimerNumero && !operacionEnCurso){
+        numeroSigno = parseFloat(PrimerNumero);
+        if(numeroSigno<0){
+          PrimerNumero = PrimerNumero.slice(1);
+        }
+        else{
+          PrimerNumero = '-'.concat(PrimerNumero)
+        }
+        this.setState({
+          textoOperacion : PrimerNumero
+        });
+         
+      }
+      else{
+        numeroSigno = parseFloat(SegundoNumero);
+        if(numeroSigno<0){
+          SegundoNumero = SegundoNumero.slice(1);
+        }
+        else{
+          SegundoNumero = '-'.concat(SegundoNumero)
+        }
+        let posicion = this.state.textoOperacion.indexOf(operacionEnCurso)
+        this.setState({
+          textoOperacion : this.state.textoOperacion.slice(0,posicion+2) + SegundoNumero,
+        });
+
+        
+      }
+    }
+  }
+
+
+
+  render(){
+
+    let operaciones = ['+','-','x','/'];
```

Comment:
Define all inmutable variables as `const`.

```suggestion
    const operaciones = ['+','-','x','/'];
   
```

Also, move them to a separate `constants.js` file